runtime/compiler/optimizer/FearPointAnalysis.cpp

```diff
@@ -34,10 +34,10 @@
 #include "infra/BitVector.hpp"
 #include "infra/Checklist.hpp"
 
-bool TR_FearPointAnalysis::virtualGuardsKillFear()
+bool TR_FearPointAnalysis::virtualGuardsKillFear(TR::Compilation *comp)
    {
    static bool kill = (feGetEnv("TR_FPAnalaysisGuardsDoNotKillFear") == NULL);
-   return kill;
+   return kill && !comp->getMethodSymbol()->hasVectorAPI();
```

Comment:
We should probably check for the `disableVectorAPIExpansion` option here too, like we do for `IfVectorAPI`